frontend/src/metabase/common/components/EllipsifiedPath/EllipsifiedPath.tsx

```diff
@@ -0,0 +1,52 @@
+import { useAreAnyTruncated } from "metabase/hooks/use-is-truncated";
+import { Tooltip } from "metabase/ui";
+
+import S from "./EllipsifiedPath.module.css";
+
+type EllipsifiedPathProps = { items: string[]; tooltip: string };
+
+/**
+ * Displays a path such as "Collection / Subcollection / Subsubcollection /
+ * Parent Collection".
+ *
+ * If the path is too long to fit, some items may be truncated, like this:
+ * "Collection / Subcollec... / Subsub... / Parent Collection".
+ *
+ * The first and last item are never truncated.
+ *
+ * A tooltip is shown if any items are truncated.
+ */
+export const EllipsifiedPath = ({ items, tooltip }: EllipsifiedPathProps) => {
+  const { areAnyTruncated, ref } = useAreAnyTruncated<HTMLDivElement>();
+
+  return (
+    <Tooltip label={tooltip} disabled={!areAnyTruncated} multiline maw="20rem">
+      <div className={S.path}>
+        {items.length > 1 && (
+          <div className={S.dots}>
+            … <div className={S.slash}>/</div>
+          </div>
+        )}
+        {items.map((item, index) => {
+          const key = `${item}${index}`;
+          return (
+            <>
+              <div
+                key={key}
+                ref={el => el && ref.current.set(key, el)}
+                className={S.item}
+              >
+                {item}
+              </div>
+              {index < items.length - 1 && (
+                <div key={`${key}-sep`} className={S.slash}>
+                  /
+                </div>
+              )}
+            </>
+          );
+        })}
```

Comment:
nit, if you generate the key inline you can eliminate some lines here
```suggestion
        {items.map((item, index) => (
            <>
              <div
                key={`${item}${index}`}
                ref={el => el && ref.current.set(key, el)}
                className={S.item}
              >
                {item}
              </div>
              {index < items.length - 1 && (
                <div key={`${key}-sep`} className={S.slash}>
                  /
                </div>
              )}
            </>
          ))}
```